src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultViewGenerator.java

```diff
@@ -61,25 +64,78 @@ public CaseSearchResultView execute(String caseTypeId,
-        caseSearchResult.getCases().forEach(caseDetails ->
-            items.add(buildSearchResultViewItem(caseDetails, caseTypeDefinition, searchResultDefinition)));
-
+        caseSearchResult.getCases().forEach(caseDetails -> {
+            filterResultsByAuthorisationAndUserRole(useCase, caseDetails, caseTypeId, requestedFields);
+            items.add(buildSearchResultViewItem(caseDetails, caseTypeDefinition, searchResultDefinition));
+        });
         return items;
     }
 
-    private List<SearchResultViewHeaderGroup> buildHeaders(String caseTypeId,
-                                                           String useCase,
-                                                           CaseSearchResult caseSearchResult,
-                                                           List<String> requestedFields) {
+    private CaseDetails filterResultsByAuthorisationAndUserRole(String useCase, CaseDetails caseDetails, String caseTypeId, List<String> requestedFields) {
+        CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
+        caseDetails.getData().entrySet().removeIf(
+            caseField -> {
+                Optional<CaseFieldDefinition> caseFieldDefinition = caseTypeDefinition.getCaseField(caseField.getKey());
+                return !filterResultsBySearchResultsDefinition(useCase, caseTypeId, requestedFields, caseField.getKey())
+                    || !filterResultsByAuthorisationAccessOnField(caseFieldDefinition.get());
+            });
+        return caseDetails;
+    }
+
+
+    private Boolean filterResultsBySearchResultsDefinition(String useCase, String caseTypeId, List<String> requestedFields, String caseFieldId) {
+        Set<String> roles = userRepository.getUserRoles();
+        CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
+        SearchResult searchResultDefinition = searchResultDefinitionService.getSearchResultDefinition(caseTypeDefinition, useCase, requestedFields);
+        HashMap<String, String> searchFields = getSearchResultDefinitionFieldUserRoleAndField(searchResultDefinition);
+
+        if (useCase != null) {
+            if (!searchFields.containsKey(caseFieldId)) {
+                return false;
+            }
+            String role = searchFields.get(caseFieldId);
+            if (role != null && !roles.contains(role)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private Boolean filterResultsByAuthorisationAccessOnField(CaseFieldDefinition caseFieldDefinition) {
+        if (!caseFieldDefinition.isMetadata()) {
+            return userRepository.getUserRoles().stream()
+                .anyMatch(role -> caseFieldDefinition.getAccessControlListByRole(role).map(AccessControlList::isRead).orElse(false));
+        }
+        return true;
+    }
+
+    private Boolean filterResultsBySecurityClassification(CaseFieldDefinition caseFieldDefinition,
```

Comment:
just to double check, do we have a test proving this works for nested fields specified with a path?